ui/page/governance/proposal_details_page.go

```diff
@@ -163,6 +169,11 @@ func (pg *ProposalDetails) listenForSyncNotifications() {
 // Part of the load.Page interface.
 func (pg *ProposalDetails) OnNavigatedFrom() {
 	pg.ctxCancel()
+
+	if pg.PoliteiaNotifCh != nil {
+		close(pg.PoliteiaNotifCh)
+	}
+	pg.WL.MultiWallet.Politeia.RemoveNotificationListener(ProposalDetailsPageID)
```

Comment:
Remove before closing the channel. Also makes sense to set `pg.PoliteiaNotifCh` to nil, wouldn't be receiving any notifications anymore until you recall `MultiWallet.Politeia.AddNotificationListener`.

Reply:
Similar considerations for other uses of notification listeners.